### PR TITLE
[minor] Capability to provision and deprovision fyre clusters against different sites

### PR DESCRIPTION
**Pull request by alequint**
So far, only default (svl) was used. Now we can specify an alternative site (like rtp). This is important as teams are getting quota in different sites and must be able to use automation to create/destroy clusters from there.

Update: fyre cluster successfully provisioned in rtp side of Fyre:

image

### CODE REVIEW

File: ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml [hunk 1]
```diff
@@ -80,7 +81,7 @@
 # -----------------------------------------------------------------------------
 - name: "fyre : Follow deployment status (2 minute intervals)"
   uri:
-    url: "https://ocpapi.svl.ibm.com/v1/ocp/{{ cluster_name }}/status"
+    url: "https://ocpapi.svl.ibm.com/v1/ocp/{{ cluster_name }}/status?site={{ fyre_site }}"
```
Comment (Aymen-Ben-S) [inline ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml:84]: Status doesn't either 👍

File: ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml [hunk 2]
```diff
@@ -39,7 +40,7 @@
 # -----------------------------------------------------------------------------
 - name: "fyre : Check if cluster already exists"
   uri:
-    url: "https://ocpapi.svl.ibm.com/v1/check_hostname/{{ cluster_name }}"
+    url: "https://ocpapi.svl.ibm.com/v1/check_hostname/{{ cluster_name }}?site={{ fyre_site }}"
```
Comment (Aymen-Ben-S) [inline ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml:43]: Check hostname doesn't need it.

File: ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml [hunk 3]
```diff
@@ -41,7 +42,7 @@
     - _cluster_exist.json is defined
     - _cluster_exist.json.owning_user is defined # when there's cluster owner, it means the cluster exists thus we can deprovision it
   uri:
-    url: "https://ocpapi.svl.ibm.com/v1/ocp/{{cluster_name}}"
+    url: "https://ocpapi.svl.ibm.com/v1/ocp/{{ cluster_name }}?site={{ fyre_site }}"
```
Comment (Aymen-Ben-S) [inline ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml:45]: To delete also.

File: ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml [hunk 4]
```diff
@@ -16,12 +16,13 @@
...
       - "Username ..................... {{ fyre_username }}"
       - "Password ..................... *****************"
+      - "Site ......................... {{ fyre_site }}"
 
 # 2. Determine whether cluster exists
 # -----------------------------------------------------------------------------
 - name: "quickburn : Check if cluster already exists"
   uri:
-    url: "https://ocpapi.svl.ibm.com/v1/check_hostname/{{ cluster_name }}"
+    url: "https://ocpapi.svl.ibm.com/v1/check_hostname/{{ cluster_name }}?site={{ fyre_site }}"
```
Comment (Aymen-Ben-S) [inline ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml:25]: Site may not be necessary for check_hostname.

**Review by Aymen-Ben-S — COMMENTED** [review]
Some calls don't need to know the site as the fyre hostname domain is the same for both sites.

File: ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml (hunk 4 above)
Comment (alequint) [reply ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml:25]: @Aymen-Ben-S I thought the same till the guy from Fyre suggested the approach above for GETS and use the side in the payload for POST and PUT. It seems svl url is the same regardless the side we want to provision.

File: ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml (hunk 3 above)
Comment (alequint) [reply ibm/mas_devops/roles/ocp_deprovision/tasks/providers/fyre.yml:45]: Same as https://github.com/ibm-mas/ansible-devops/pull/1085/files#r1383330877

image

File: ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml (hunk 2 above)
Comment (alequint) [reply ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml:43]: Same as:

image

File: ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml (hunk 1 above)
Comment (alequint) [reply ibm/mas_devops/roles/ocp_provision/tasks/providers/fyre.yml:84]: Same as https://github.com/ibm-mas/ansible-devops/pull/1085/files#r1383334103

**Pull request merged**